File: app/src/main/java/grey/smarthouse/ui/mainScreen/MainActivity.java
package grey.smarthouse.ui.mainScreen;

import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabLayout;

import grey.smarthouse.R;
import grey.smarthouse.model.App;
import grey.smarthouse.retrofit.Requests;
import grey.smarthouse.services.NetService;
import grey.smarthouse.ui.SingleFragmentActivity;
import io.reactivex.Observable;

/**
 * Created by GREY on 26.05.2018.
 */

public class MainActivity extends SingleFragmentActivity {



    private ViewPager mViewPager;
    private final static String NOTIFICATION_FLAG =  "nFlag";
    private final static String NOTIFICATION_TEMP =  "nTemp";
    @Override
    protected Fragment createFragment() {
        //return SensorListFragment.newInstance(mPage);
        return null;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mViewPager =findViewById(R.id.view_pager);
        FragmentManager fm = getSupportFragmentManager();
        mViewPager.setAdapter(new ViewPagerAdapter(fm));

        TabLayout tabLayout = findViewById(R.id.tabs);
        tabLayout.setupWithViewPager(mViewPager);
        Requests.RetrofitInit(App.getApp().mDeviceURL);
        Intent intent = new Intent(getApplicationContext(), NetService.class);
        intent.putExtra(NOTIFICATION_FLAG,  App.getApp().mIsNotifOn);
        intent.putExtra(NOTIFICATION_TEMP,  App.getApp().mNotifTemp);
        startService(intent);
        Observable<String> observable = Observable.just("one",
                "two", "three");


//       observable.subscribe(x->Log.d("RX","next"),
//               e->e.printStackTrace(),
//               ()->Log.d("RX","complete"),
//               d->Log.d("RX","sub"));


    }
}

File: app/src/main/java/grey/smarthouse/ui/mainScreen/MainActivity.kt
package grey.smarthouse.ui.mainScreen

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.observe
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import grey.smarthouse.App
import grey.smarthouse.R
import grey.smarthouse.data.remote.Requests
import grey.smarthouse.ui.mainScreen.locations.LocationDialog
import grey.smarthouse.ui.mainScreen.locations.LocationVM
import grey.smarthouse.ui.mainScreen.relays.RelaysVM
import grey.smarthouse.ui.mainScreen.sensors.SensorDialog
import grey.smarthouse.ui.mainScreen.sensors.SensorsVM
import grey.smarthouse.ui.relaySettingsScreen.RelaySettingsActivity
import kotlinx.android.synthetic.main.activity_main.*
import javax.inject.Inject

/**
 * Created by GREY on 26.05.2018.
 */

class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var relaysVM: RelaysVM

    @Inject
    lateinit var locationsVM: LocationVM

    @Inject
    lateinit var sensorsVM: SensorsVM

    private lateinit var mViewPager: ViewPager


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        locationsVM = (application as App).appComponent.getLocationVM()
        sensorsVM = (application as App).appComponent.getSensorsVM()
        relaysVM = (application as App).appComponent.getRelaysVM()
        setContentView(R.layout.activity_main)
        mViewPager = view_pager
        val fm = supportFragmentManager
        mViewPager.adapter = ViewPagerAdapter(fm, 1)
        val tabLayout = findViewById<TabLayout>(R.id.tabs)
        tabLayout.setupWithViewPager(mViewPager)
        Requests.retrofitInit(App.app.mDeviceURL)

        relaysVM.openRelayEvent.observe(this) {
            it?.let {
                openRelaySettings(it)
            }
        }

        locationsVM.editLocationEvent.observe(this) {
            it.let {
                val args = Bundle()
                args.putInt("id", it)
                LocationDialog().apply {
                    arguments = args
                    show(supportFragmentManager, "locationDialog")
                }
            }
        }

        sensorsVM.editSensorEvent.observe(this) {
            val args = Bundle()
            args.putInt("num", it)
            SensorDialog().apply {
                arguments = args
                show(supportFragmentManager, "sensorDialog")
            }
        }
    }

    companion object {
        private val NOTIFICATION_FLAG = "nFlag"
        private val NOTIFICATION_TEMP = "nTemp"
    }

    private fun openRelaySettings(num: Int) {
        val intent = RelaySettingsActivity.NewIntent(this, num)
        startActivity(intent)
    }
}
